### PR TITLE
[refactor] Validate item names/ids without allocating Vec

### PR DESCRIPTION
**Pull request by suiwenfeng**
Signed-off-by: Wenfeng Sui <suiwenfeng@qq.com>

### CODE REVIEW

File: config/src/client.rs [hunk 1]
```diff
@@ -142,37 +142,45 @@ impl ConfigurationProxy {
...
-                ));
-            }
-            // TODO: this is neither robust, nor useful. This should be enforced as a `FromStr` implementation.
-            if split_api_url[0] != "http" {
-                eyre::bail!(ConfigError::ProxyBuildError(
-                    "`TORII_API_URL` string: `{api_url}` only supports the `HTTP` protocol currently".to_owned()
-                ));
+            let split = api_url.rsplit_once("://");
+            match split {
+                Some((protocal, _)) => {
```
Comment (6r1d) [inline config/src/client.rs:147]: `protocal`, not a "protocol"? Same applies at 149, 167, 168

File: data_model/tests/data_model.rs [hunk 2]
```diff
@@ -302,3 +302,18 @@ fn cannot_forbid_minting_on_asset_mintable_infinitely() -> Result<(), ParseError
     assert!(definition.forbid_minting().is_err());
     Ok(())
 }
+
+#[test]
+fn account_id_parsing() -> Result<(), ParseError> {
+    // `AccountId` should have format `name@domain_name`
+    let account_normal: AccountId = "test@hello".parse()?;
+    assert_eq!(account_normal.name.to_string(), String::from("test"));
```
Comment (Erigara) [inline data_model/tests/data_model.rs:310]: I would suggest to do the following: 
```suggestion
    assert_eq!(account_normal.name.as_ref(), "test");
```

File: data_model/tests/data_model.rs [hunk 3]
```diff
@@ -302,3 +302,18 @@ fn cannot_forbid_minting_on_asset_mintable_infinitely() -> Result<(), ParseError
...
+    assert_eq!(account_normal.domain_id.name.to_string(), String::from("hello"));
+
+    let account_empty: Result<AccountId, _> = "@hello".parse();
+    assert!(account_empty.is_err());
+
+    let account_invalid: Result<AccountId, _> = "testhello".parse();
+    assert!(account_invalid.is_err());
```
Comment (Erigara) [inline data_model/tests/data_model.rs:317]: Wight be redundant but we can also test for string `"@"`.

**Review by Erigara — COMMENTED** [review]
Good job!
Here few minor things you should do: 
1. We use standardize format for commit names, so you commit message should look like: `[refactor] #2675: Validate item names/ids without allocating Vec`.
2. Also we use `rustfmt` in our workflow , you can run it by following command: `cargo +nightly fmt --all`

File: data_model/tests/data_model.rs (hunk 3 above)
Comment (suiwenfeng) [reply data_model/tests/data_model.rs:317]: i reopened a pr for some reason, you have to check this at #3004 now.

**Pull request closed without merging**